tests/Support/SupportCollectionTest.php

```diff
@@ -2519,6 +2520,35 @@ public function testGetNestedValue()
         $this->assertEquals('Book 1', $collection->get('foo.books.0'));
         $this->assertEquals('Todo 2', $collection->get('foo.todos.second'));
     }
+
+    public function testHasReturnsValidResults()
+    {
+        $collection = new Collection(['foo' => 'one', 'bar' => 'two', 1 => 'three']);
+        $this->assertTrue($collection->has('foo'));
+        $this->assertTrue($collection->has('foo', 'bar', 1));
+        $this->assertFalse($collection->has('foo', 'bar', 1, 'baz'));
+        $this->assertFalse($collection->has('baz'));
+    }
+
+    public function testPutAddsItemToCollection()
+    {
+        $collection = new Collection();
+        $this->assertSame([], $collection->toArray());
+        $collection->put('foo', 1);
+        $this->assertSame(['foo' => 1], $collection->toArray());
+        $collection->put('bar', ['nested' => 'two']);
+        $this->assertSame(['foo' => 1, 'bar' => ['nested' => 'two']], $collection->toArray());
+        $collection->put('foo', 3);
+        $this->assertSame(['foo' => 3, 'bar' => ['nested' => 'two']], $collection->toArray());
+    }
+
+    public function testItThrowsExceptionWhenTryingToAccessNoProxyProperty()
+    {
+        $collection = new Collection();
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage('Property [foo] does not exist on this collection instance.');
+        $collection->foo;
```

Comment:
you are just testing if PHP works here

Reply:
Not really. This exception is thrown inside collection method and it should be tested whether it has been really thrown or not - now if you remove throwing exception from this method tests will pass and they shouldn't

Reply:
you are testing if PHP throws an exception or not when you access a non-existing property. When will this fail?

Reply:
This isn't a PHP exceptions, it's thrown from within `\Illuminate\Support\Collection::__get`. The test is correct.

Reply:
Look at implementation:

    public function __get($key)
    {
        if (! in_array($key, static::$proxies)) {
            throw new Exception("Property [{$key}] does not exist on this collection instance.");
        }

        return new HigherOrderCollectionProxy($this, $key);
    }

so I'm testing if this `if` is added and working as expected. Now there is no such test. So I'm not testing PHP, I'm testing whether this code exists and is executed as expected in this method

Reply:
Well yes but I still don't see the point. If this changes, other tests which use `$foo->bar` will fail as well so it is not going to be missed.